packages/de1/src/de1.ts

```diff
@@ -75,11 +75,58 @@ export default class DE1 {
     return level;
   }
 
+  async getMixTemp(): Promise<Value<"shot">["mixTemp"]> {
+    const { mixTemp } = await this.get("shot");
+    return mixTemp;
+  }
+
+  async getHeadTemp(): Promise<Value<"shot">["headTemp"]> {
+    const { headTemp } = await this.get("shot");
+    return headTemp;
+  }
+
+  async getSteamTemp(): Promise<Value<"shot">["steamTemp"]> {
+    const { steamTemp } = await this.get("shot");
+    return steamTemp;
+  }
+
+  async getShotSettings(): Promise<Value<"shotSettings">> {
+    return await this.get("shotSettings");
+  }
+
+  async setShotSettings(
+    settings: Partial<Value<"shotSettings">>,
+  ): Promise<void> {
+    await this.set("shotSettings", settings);
+  }
+
+  async getVersions(): Promise<Value<"versions">> {
+    return await this.get("versions");
+  }
+
+  async get<N extends Name>(name: N): Promise<Value<N>> {
+    return await this.machine.read(name);
+  }
+
+  async set<N extends Name>(name: N, value: Partial<Value<N>>): Promise<void> {
+    const mergedValue = await this.mergeCurrentValue(name, value);
+    await this.machine.write(name, mergedValue);
+  }
+
   on<N extends Name>(name: N, listener: Listener<N>): void {
     this.machine.on(name, listener);
   }
 
   off<N extends Name>(name: N, listener: Listener<N>): void {
     this.machine.off(name, listener);
   }
+
+  private async mergeCurrentValue<N extends Name>(
+    name: N,
+    value: Partial<Value<N>>,
+  ): Promise<Value<N>> {
+    if (typeof value !== "object" || value === null) return value;
+    const currentValue = await this.machine.read(name);
+    return Object.assign({}, currentValue, value);
```

Comment:
I prefer this syntax, but up to you:

```suggestion
    return {...currentValue, ...value};
```

Reply:
No you're absolutely right!